### PR TITLE
docs: add Casbin plug-in blog

### PR DESCRIPTION
**Pull request by SylviaBABY**

### CODE REVIEW

File: _posts/blog/zh-CN/use-authz-casbin.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
+---
+title: "在 Apache APISIX 中使用 Casbin 进行授权"
+avatar: "https://avatars.githubusercontent.com/u/23514812?v=4"
+author: "赵若妃"
+href: "https://github.com/Serendipity96"
```
Comment (yzeng25) [inline _posts/blog/zh-CN/use-authz-casbin.md:5]: avatar: "https://avatars.githubusercontent.com/u/30819887?v=4"
author: "屠正松"
href: "https://github.com/tzssangglass"

File: _posts/blog/zh-CN/use-authz-casbin.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+### 创建一个模型
+
+authz-casbin 插件使用三个参数来进行授权：subject、object 和 action。subject 是用户名，代指请求中的用户；object 是将要被访问的 URL 链接也就是将被访问的资源；action 是请求授权的行为，比如读操作（read）或者是写操作（write）。假如，我们想要创建一个模型来访问三个资源：/，/res1，/res2，我们想要一个类似于这样的模型：
+
+![authz-casbin model](https://static.apiseven.com/202108/1630655235372-ec729a09-10c3-4209-87c4-2b7455bfdf14.jpeg)
+
+
+在这个模型中，所有的用户，例如 Jack，可以访问主页面（/）。而像 Alice 和 Bob 具有管理员权限的用户则可以访问所有的页面和资源（/res1，/res2，/）。这样，我们就需要来限制没有管理员权限的用户只能使用 GET 请求方法访问特定的资源。所需要的模型如下：
+
+```
```
Comment (yzeng25) [inline _posts/blog/zh-CN/use-authz-casbin.md:38]: code type can be found in the doc below
https://tfzcfxawmk.feishu.cn/docs/doccn4cMri7llyONgW2FNT34FHf

For example, line 38 should change to "```shell"

File: _posts/blog/zh-CN/use-authz-casbin.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+[policy_effect]
+e = some(where (p.eft == allow))
+[matchers]
+m = (g(r.sub, p.sub) || keyMatch(r.sub, p.sub)) && keyMatch(r.obj, p.obj) && keyMatch(r.act, p.act)
+```
+
+### 创建一条策略
+从上述的例子来看，策略应该像是这样的：
+
+```
```
Comment (yzeng25) [inline _posts/blog/zh-CN/use-authz-casbin.md:54]: ditto

File: _posts/blog/zh-CN/use-authz-casbin.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+1. `(g(r.sub, p.sub) || keyMatch(r.sub, p.sub))`：请求里的 subject 和策略里的 subject 有着相同的角色或者请求里的 subject 和策略里的 subject 可以通过内置的方法 keyMatch 匹配。keyMatch 作为 Lua Casbin 的内置函数，相关的描述以及更多的函数可跳转 lua-casbin (https://github.com/casbin/lua-casbin/blob/master/src/util/BuiltInFunctions.lua)。
+2. `keyMatch(r.obj, p.obj)`：请求里的 object 和策略里的 object 可相互匹配（代指 URL 链接）。
+3. `keyMatch(r.act, p.act)`：请求里的 action 和策略里的 action 可相互匹配（代指 HTTP 请求方法）。
+
+### 在路由上使用插件
+
+一旦你创建了模型和策略，你可以使用 APISIX Admin API 在路由上使用。若想使用，你可以模型和策略的文件路径：
+
+```json
```
Comment (yzeng25) [inline _posts/blog/zh-CN/use-authz-casbin.md:71]: ditto, this should be  "```shell", double check with https://tfzcfxawmk.feishu.cn/docs/doccn4cMri7llyONgW2FNT34FHf

File: _posts/blog/zh-CN/use-authz-casbin.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        },
+        "type": "roundrobin"
+    },
+    "uri": "/*"
+}'
+```
+
+在这里，username 是传递到 subject 里的用户名。例如，你可以设置"username":"user"来把你定义的"user":"alice"传递到 username ，让 username 成为 Alice。同样，你可以将模型和策略直接放到里面：
+
+```json
```
Comment (yzeng25) [inline _posts/blog/zh-CN/use-authz-casbin.md:93]: ditto

File: _posts/blog/zh-CN/use-authz-casbin.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    },
+    "uri": "/*"
+}'
+```
+
+### 利用全局模型/策略使用插件
+
+在一些情形中，你可能想要在多个路由中使用相同的模型和策略，你可以首先发送一个 PUT 请求将模型和策略的配置发送到插件的元数据：
+
+```
```
Comment (yzeng25) [inline _posts/blog/zh-CN/use-authz-casbin.md:128]: ditto

File: _posts/blog/zh-CN/use-authz-casbin.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+"policy": "p, *, /, GET
+p, admin, *, *
+g, alice, admin
+g, bob, admin"
+}'
+```
+
+然后，你需要使用 Admin API 来发送请求使得多个路由使用相同的模型/策略配置；
+
+```json
```
Comment (yzeng25) [inline _posts/blog/zh-CN/use-authz-casbin.md:150]: ditto, it should be shell instead of json

**Review by yzeng25 — CHANGES_REQUESTED** [review]

File: _posts/blog/zh-CN/use-authz-casbin.md (hunk 1 above)
Comment (SylviaBABY) [reply _posts/blog/zh-CN/use-authz-casbin.md]: done

File: _posts/blog/zh-CN/use-authz-casbin.md (hunk 2 above)
Comment (SylviaBABY) [reply _posts/blog/zh-CN/use-authz-casbin.md]: done

File: _posts/blog/zh-CN/use-authz-casbin.md (hunk 3 above)
Comment (SylviaBABY) [reply _posts/blog/zh-CN/use-authz-casbin.md]: done

File: _posts/blog/zh-CN/use-authz-casbin.md (hunk 4 above)
Comment (SylviaBABY) [reply _posts/blog/zh-CN/use-authz-casbin.md]: done

File: _posts/blog/zh-CN/use-authz-casbin.md (hunk 5 above)
Comment (SylviaBABY) [reply _posts/blog/zh-CN/use-authz-casbin.md]: done

File: _posts/blog/zh-CN/use-authz-casbin.md (hunk 6 above)
Comment (SylviaBABY) [reply _posts/blog/zh-CN/use-authz-casbin.md]: done

File: _posts/blog/zh-CN/use-authz-casbin.md (hunk 7 above)
Comment (SylviaBABY) [reply _posts/blog/zh-CN/use-authz-casbin.md]: done

**Review by yzeng25 — APPROVED** [review]

**Pull request merged**